substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrFeature.java

```diff
@@ -163,4 +174,31 @@ public void beforeCompilation(BeforeCompilationAccess a) {
             JfrTraceId.assign(clazz, hub.getTypeID() + 1);
         }
     }
+
+    @Override
+    public void duringAnalysis(DuringAnalysisAccess access) {
+        Class<?> eventClass = access.findClassByName("jdk.internal.event.Event");
+        if (eventClass != null && access.isReachable(eventClass)) {
+            Set<Class<?>> s = access.reachableSubtypes(eventClass);
+            for (Class<?> c : s) {
+                if (c.getCanonicalName().equals("jdk.jfr.Event")
+                        || c.getCanonicalName().equals("jdk.internal.event.Event")
+                        || c.getCanonicalName().equals("jdk.jfr.events.AbstractJDKEvent")) {
+                    continue;
+                }
+                fieldRegistration.computeIfAbsent(c, this::registerEventHandler);
```

Comment:
`RuntimeReflectionSupport` already uses a set internally for deduplication. So, you shouldn't need `ConcurrentHashMap fieldRegistration`.